parcels/particlefile/baseparticlefile.py

```diff
@@ -92,12 +94,16 @@ def __init__(self, name, particleset, outputdt=np.infty, write_ondelete=False, c
             self.time_origin = self.particleset.time_origin
             self.lonlatdepth_dtype = self.particleset.collection.lonlatdepth_dtype
             self.var_names = []
+            self.var_dtypes = []
             self.var_names_once = []
+            self.var_dtypes_once = []
             for v in self.particleset.collection.ptype.variables:
-                if v.to_write == 'once':
+                if v.to_write == 'once' and self.write_ondelete is False:
                     self.var_names_once += [v.name]
-                elif v.to_write is True:
+                    self.var_dtypes_once += [v.dtype.__name__]
+                elif v.to_write is True or v.to_write == 'once':
                     self.var_names += [v.name]
+                    self.var_dtypes += [v.dtype.__name__]
```

Comment:
Same as above, why not store the object?
```suggestion
                    self.var_dtypes += [v.dtype]
```